be/src/exec/spill/mem_table.h

```diff
@@ -0,0 +1,71 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <utility>
+
+#include "column/column_helper.h"
+#include "column/vectorized_fwd.h"
+#include "common/status.h"
+#include "exec/sorting/sort_permute.h"
+#include "exec/sorting/sorting.h"
+#include "exprs/expr_context.h"
+#include "runtime/mem_tracker.h"
+#include "runtime/runtime_state.h"
+
+namespace starrocks {
+using FlushCallBack = std::function<Status(const ChunkPtr&)>;
+
+//  This component is the intermediate buffer for our spill data, which may be ordered or unordered,
+// depending on the requirements of the upper layer
+
+// usage:
+//
+// auto mem_table = create();
+// while (!mem_table->is_full()) {
+//     mem_table->append(next_chunk());
+// }
+// mem_table->done();
+// mem_table->flush();
+
+class SpilledMemTable {
+public:
+    SpilledMemTable(RuntimeState* state, size_t max_buffer_size, MemTracker* parent)
```

Comment:
`parent` is not used here.

Reply:
yes. but it will be used in future. (we may want to know all of the memory table memory usage)